opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -236,6 +236,9 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             if error_message is not None:
                 logger.warning("%s in attribute value sequence", error_message)
                 return
+            else:
+                # convert value : sequence into immutable tuple lest client changes it.
+                value_tuple = tuple(value)
```

Comment:
I think you want this instead:

```suggestion
                value = tuple(value)
```

Reply:
oops yes. changed to tuple conversion in-situ.